src/components/client.ts

```diff
@@ -38,18 +40,21 @@ export class <%= className %> {
     if (options.transport === undefined || options.transport.type === undefined) {
       throw new Error("Invalid constructor params");
     }
-    const {type, host, port} = options.transport;
+    const {type, host, port, protocol, postMessageType} = options.transport;
     let path = options.transport.path || "";
     if(path && path[0] !== "/") {
         path = "/" + path;
     }
     switch (type) {
       case 'http':
       case 'https':
-        this.transport = new HTTPTransport(type + "://" + host + ":" + port + path)
+        this.transport = new HTTPTransport((protocol || type) + "://" + host + ":" + port + path);
```

Comment:
why protocol here?

Reply:
I think we should get rid of protocol and type and create a mapping structure. Then you can change the switch on type to simply retrieve the translation from a map. something like 

```ts
type ProtocolNames = "ws" | "http" | "https" | "postmessageiframe" | "postmessagewindow";
type NameMapping = Record<ProtocolNames,string>
const protocolNaming: NameMapping = {
  "ws": "websocket",
  "http":"http",
  "https": "https",
  "postmessageiframe": "postmessage",
  "postmessagewindow": "postmessagew"
}
const protocolType = "ws"
protocolNaming[protocolType]
```

Reply:
I want the user to be able to override protocol

Reply:
`http://localhost:3000` vs `https://sig.tools`


there is no way to define this any other way
![image](https://user-images.githubusercontent.com/364566/85657324-be122100-b666-11ea-9225-8f7bdeb22c32.png)

Reply:
using `protocol` here would follow the nodejs URL parsing API of `.protocol`: https://nodejs.org/api/url.html#url_url_protocol